### PR TITLE
Improvements to aydin docstrings

### PR DESCRIPTION
**Pull request by royerloic**
Improved a lot of doc-strings, particularly for feature gen and FGR backends.

### CODE REVIEW

**Review by AhmetCanSolak — COMMENTED** [review]
Thank you @royerloic ! I dropped a few comments to discuss if we can add further more info on certain places. After discussing these I think we can merge this. After the merge I will be making a follow-up PR to address changes on our documentation too and will update documentation webpage accordingly.

File: aydin/it/fgr.py [hunk 1]
```diff
@@ -43,8 +43,10 @@ def __init__(
         balance_training_data : bool
             Limits number training entries per target
             value histogram bin.
+            (advanced)
```
[the remark this answers is not in the document]
Comment (royerloic) [reply aydin/it/fgr.py:46]: Good idea! It is actually an important feature!

File: aydin/features/groups/uniform.py [hunk 2]
```diff
@@ -27,24 +28,79 @@ def _remove_duplicates(seq):
...
+        scale_one_width : int
+            Width of the scale one features
+            (advanced)
+        include_scale_one : bool
+            True or False, argument to set decision
+            on inclusion of scale-one-features
+            (advanced)
+        include_fine_features : bool
+            True or False, argument to set decision
+            on inclusion of fine features
```
[the remark this answers is not in the document]
Comment (royerloic) [reply aydin/features/groups/uniform.py:84]: Done.

File: aydin/regression/perceptron.py [hunk 3]
```diff
@@ -20,8 +20,15 @@
 from aydin.util.tf.device import get_best_device_name
 
 
-class NNRegressor(RegressorBase):
-    """standard perceptron-like neural network regressor."""
+class PerceptronRegressor(RegressorBase):
```
[the remark this answers is not in the document]
Comment (royerloic) [reply aydin/regression/perceptron.py:23]: haha. Please check I did not forget to rename something else...

**Review by AhmetCanSolak — APPROVED** [review]
Thank you for the updates. Looks great. Now merging!

**Pull request merged**